apps/sms/views/conversation/js/compose.js

```diff
@@ -656,9 +656,12 @@ var Compose = (function() {
           }
         });
 
-      //'pick cancelled' is the error thrown when the pick activity app is
-      // canceled normally
-      } else if (errId !== 'ActivityCanceled' && errId !== 'pick cancelled') {
+      // Some possible 'activity cancel' error name:
+      // 'pick cancelled': error thrown when the pick activity canceled in most
+      //   of the apps.
+      // 'cancelled': error thrown when the pick activity canceled in contacts.
+      // 'ActivityCanceled': error thrown when pick activity canceled from menu.
+      } else if (!/cancel/i.test(errId)) {
```

Comment:
There's something really messy about our cancel :/  The error message is defined by pick apps but not sync with the unique 'ActivityCanceled', and I've seen pick cancelled(most of the apps)/pick 'canceled' (wallpaper and some other apps)/cancelled (contacts only). Maybe we should file a bug to unify the naming of the activity canceled.